Fix Hexagon bindgen linter warnings

Fix the following bindgen warnings:
```
Missing type comment at <rizin/librz/asm/arch/hexagon/hexagon.h:151:10> (token is not a comment)
Mismatched function annotation for hex_const_ext_free at <rizin/librz/asm/arch/hexagon/hexagon_arch.h:69:13> : None (was RZ_NULLABLE at <rizin/librz/asm/arch/hexagon/hexagon_arch.c:226:13>
Missing type comment at <rizin/librz/asm/arch/hexagon/hexagon_arch.c:787:59> (token is not a comment)
Error: Process completed with exit code 3.
```
See https://github.com/rizinorg/rizin/runs/8289833908?check_suite_focus=true

--- a/librz/asm/arch/hexagon/hexagon.h
+++ b/librz/asm/arch/hexagon/hexagon.h
@@ -3,7 +3,7 @@
 
 // LLVM commit: 96e220e6886868d6663d966ecc396befffc355e7
 // LLVM commit date: 2022-01-05 11:01:52 +0000 (ISO 8601 format)
-// Date of code generation: 2022-08-07 13:35:47-04:00
+// Date of code generation: 2022-09-12 14:26:04-04:00
 //========================================
 // The following code is generated.
 // Do not edit. Repository of code generator:
@@ -148,7 +148,7 @@ typedef struct {
  */
 typedef struct {
 	HexPkt pkts[HEXAGON_STATE_PKTS]; // buffered instructions
-	RzList *const_ext_l; // Constant extender values.
+	RzList /*<HexConstExt *>*/ *const_ext_l; // Constant extender values.
 	RzAsm rz_asm; // Copy of RzAsm struct. Holds certain flags of interesed for disassembly formatting.
 	RzConfig *cfg;
 	RzPVector /*<RzAsmTokenPattern *>*/ *token_patterns; ///< PVector with token patterns. Priority ordered.

--- a/librz/asm/arch/hexagon/hexagon_arch.c
+++ b/librz/asm/arch/hexagon/hexagon_arch.c
@@ -784,7 +784,7 @@ static inline bool imm_is_scaled(const HexOpAttr attr) {
  * \param addr The address of the instruction which gets the constant extender applied.
  * \return HexConstExt* A const. ext., if there is one which should be applied on the instruction at addr. Otherwise NULL.
  */
-static HexConstExt *get_const_ext_from_addr(const RzList *ce_list, const ut32 addr) {
+static HexConstExt *get_const_ext_from_addr(const RzList /*<HexConstExt *>*/ *ce_list, const ut32 addr) {
 	HexConstExt *ce = NULL;
 	RzListIter *iter = NULL;
 	rz_list_foreach (ce_list, iter, ce) {

--- a/librz/asm/arch/hexagon/hexagon_arch.h
+++ b/librz/asm/arch/hexagon/hexagon_arch.h
@@ -66,7 +66,7 @@ RZ_API HexInsn *hexagon_alloc_instr();
 RZ_API void hex_insn_free(RZ_NULLABLE HexInsn *i);
 RZ_API HexInsnContainer *hexagon_alloc_instr_container();
 RZ_API void hex_insn_container_free(RZ_NULLABLE HexInsnContainer *c);
-RZ_API void hex_const_ext_free(HexConstExt *ce);
+RZ_API void hex_const_ext_free(RZ_NULLABLE HexConstExt *ce);
 RZ_API HexState *hexagon_get_state();
 RZ_API void hexagon_reverse_opcode(const RzAsm *rz_asm, HexReversedOpcode *rz_reverse, const ut8 *buf, const ut64 addr);
 RZ_API ut8 hexagon_get_pkt_index_of_addr(const ut32 addr, const HexPkt *p);

--- a/librz/asm/p/asm_hexagon.c
+++ b/librz/asm/p/asm_hexagon.c
@@ -3,7 +3,7 @@
 
 // LLVM commit: 96e220e6886868d6663d966ecc396befffc355e7
 // LLVM commit date: 2022-01-05 11:01:52 +0000 (ISO 8601 format)
-// Date of code generation: 2022-04-17 16:44:52+02:00
+// Date of code generation: 2022-09-12 14:26:04-04:00
 //========================================
 // The following code is generated.
 // Do not edit. Repository of code generator:

